android/app/src/main/java/org/mozilla/magnet/permissions/Check.java

```diff
@@ -0,0 +1,55 @@
+package org.mozilla.magnet.permissions;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.content.pm.PackageManager;
+import android.support.v4.content.ContextCompat;
+
+/**
+ * Created by wilsonpage on 09/11/2016.
+ */
+abstract class Check {
+    private Activity mActivity;
+    private PermissionChecker mListener;
+
+    Check(PermissionChecker listener, Activity activity) {
+        mActivity = activity;
+        mListener = listener;
+    }
+
+    abstract int getId();
+    abstract void onResponse(int response);
+    abstract void check();
```

Comment:
Seems the `check` method needs to be run in the ui thread right? Since it needs usually an activity. Well, either the `check` method or all the checkers.

Maybe we can add an assertion when we create a checker object to be sure we run them in the UI thread, we can check if we are in the ui thread with:
`Looper.myLooper() == Looper.getMainLooper()`